Source/JavaScriptCore/runtime/ISO8601.cpp

```diff
@@ -530,10 +530,62 @@ std::optional<int64_t> parseUTCOffsetInMinutes(StringView string)
 template<typename CharacterType>
 static bool canBeCalendar(const StringParsingBuffer<CharacterType>& buffer)
 {
-    // https://tc39.es/proposal-temporal/#prod-Calendar
-    // Calendar :
-    //     [u-ca= CalendarName]
-    return buffer.lengthRemaining() >= 6 && buffer[0] == '[' && buffer[1] == 'u' && buffer[2] == '-' && buffer[3] == 'c' && buffer[4] == 'a' && buffer[5] == '=';
+    // https://tc39.es/proposal-temporal/#sec-temporal-parseisodatetime
+    // Step 4(a)(ii)(2)(a):
+    //  Let key be the source text matched by the AnnotationKey Parse Node contained within annotation
+    //
+    // https://tc39.es/proposal-temporal/#prod-Annotation
+    //     Annotation :::
+    //         [ AnnotationCriticalFlag[opt] AnnotationKey = AnnotationValue ]
+    //
+    //         AnnotationCriticalFlag :::
+    //             !
+    //
+    //         AnnotationKey :::
+    //             AKeyLeadingChar
+    //             AnnotationKey AKeyChar
+    //
+    //             AKeyLeadingChar :::
+    //                 LowercaseAlpha
+    //                 _
+    //
+    //             AKeyChar :::
+    //                 AKeyLeadingChar
+    //                 DecimalDigit
+    //                 -
+    //
+    //         AnnotationValue :::
+    //             AnnotationValueComponent
+    //             AnnotationValueComponent - AnnotationValue
+    //
+    //             AnnotationValueComponent :::
+    //                 Alpha AnnotationValueComponent[opt]
+    //                 DecimalDigit AnnotationValueComponent[opt]
```

Comment:
The format of these syntax is not aligned to the other code. Please check `parseTimeSpec` etc. very carefully and align it to them.